### PR TITLE
Read stratum a full packet at a time.

### PR DESCRIPTION
**Pull request by jean-m-cyr**
Use async_read_some instead of async_read_until such
that asio doesn't need to scan for any given character
and can use simple packet delineation.

### CODE REVIEW

**Comment by smurfy** [comment]
hm, not sure if this change is good, i think it will fail if only half of a message is received.
Shouldnt be there two buffers? the data arrived + the data previously arrived but not yet valid because incomplete. Only if a whole message has been received the buffer gets reduced by that message?

**Comment by jean-m-cyr** [comment]
This is standard TCP usage. TCP will not fragment a packet unless it are larger than the receiving buffer.

Was running with this for a month without issues or unnecessary poling by ASIO trying to reassemble packets from reads into smaller internal buffers

**Comment by AndreaLanfranchi** [comment]
I am not a priori against pro or against this implementation.
Problem is, for my limited perspective, we cannot determine if we're gaining something or not.

As @jean-m-cyr  said the aim is to prevent async_read_until to scan every single char but in the end we're simply moving the scan of charachters outside the boundaries of asio and doing it ourselves in the code.

Another matter is wether or not async_read_until may miss some message as per @jean-m-cyr  suspicions even if I have no empiric prove of this.

Bottom line ... **for me** async_read_until is less wordy and easier to read and mantain. This PR acts the same and is more self controlled.
